tests/probe/test_linear.py

```diff
@@ -73,39 +79,101 @@ def unpack_unravel(packed_raveled_coef, d, k, fit_intercept):
         else:
             return unpacked_unraveled_coef, packed_raveled_coef[(d * k) :]
 
-    def vectorized_objective(packed_raveled_coef):
+    def vectorized_objective(packed_raveled_coef, apply_regul=True):
         def f(prc):
             c, i = unpack_unravel(prc, d, k, fit_intercept)
-            return linearized._replace(coef=c, intercept=i).objective(X, y)
+            return LinearModel(
+                c, i, linearized.loss, None if not apply_regul else linearized.regul
+            ).objective(X, y)
 
         return np.apply_along_axis(f, axis=0, arr=packed_raveled_coef)
 
-    with np.printoptions(precision=3, suppress=True):
-        print(np.round(H := linearized.hessian(X, y), 2))
+    def vectorized_predict_proba(packed_raveled_coef):
+        def f(prc):
+            c, i = unpack_unravel(prc, d, k, fit_intercept)
+            return (
+                LinearModel(c, i, linearized.loss, linearized.regul)
+                .predict_proba(X)
+                .sum(axis=0)
+            )
+
+        return np.apply_along_axis(f, axis=0, arr=packed_raveled_coef)
+
+    with np.printoptions(precision=4, suppress=True):
+        print(H := linearized.hessian(X, y))
```

Comment:
purquoi il y a tout ces prints ?

Reply:
because the assert messages are not that clear when the tests break